proxy-node-chart/charts/metadata-controller/templates/statefulset.yaml

```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "metadata-controller.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+    helm.sh/chart: {{ include "metadata-controller.chart" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+  {{- with .Values.annotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  replicas: {{ .Values.replicaCount }}
+  serviceName: {{ include "metadata-controller.fullname" . }}
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+      app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "metadata-controller.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}-cloudhsm-client
+        image: "{{ .Values.cloudHsmClient.image.repository }}:{{ .Values.cloudHsmClient.image.tag }}"
+        imagePullPolicy: {{ .Values.cloudHsmClient.image.pullPolicy }}
+        env:
+          - name: HSM_IP
+            value: "{{ .Values.hsmIP }}"
+          - name: n3fips_user
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Release.Name }}-metadata-controller
+                key: n3fips_user
+        volumeMounts:
+          name: customer-ca-crt
+          mountPath: /opt/cloudhsm/etc
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.controller.image.repository }}:{{ .Values.controller.image.tag }}"
+        imagePullPolicy: {{ .Values.controller.image.pullPolicy }}
+        env:
+          - name: HSM_USER
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Release.Name }}-metadata-controller
+                key: HSM_USER
+          - name: HSM_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ .Release.Name }}-metadata-controller
+                key: HSM_PASSWORD
+          - name: POD_NAMESPACE
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.namespace
+        volumeMounts:
+          name: hsm-signing-cert
+          mountPath: /etc/verify-proxy-node
+      volumes:
+        - name: customer-ca-crt
```

Comment:
As above, only the cloudhsm client deployment/pod needs the customer CA cert, not the metadata controller.

Reply:
we decided to place cloudhsm client in sibling container afterall - so this volume is required for the hsm client again